File: lib/core/constants/api_constants.dart
/// API endpoints & constants for slideshow
abstract final class ApiConstants {
  /// Base URL của server standee
  static const String baseUrl = 'http://10.10.115.20:9094';

  /// WebSocket base URL
  static const String wsBaseUrl = 'ws://10.10.115.20:9094';

  /// Standee ID dùng cho playlist API
  static const String standeeId = '7bbb9af3-95a7-42af-8805-6023e60a3bdc';

  /// Relative path của API playlist
  static String get playlistPath =>
      '/api/poster/$standeeId/withActivePosters';

  /// Full URL (nếu cần debug)
  static String get playlistUrl => '$baseUrl$playlistPath';

  /// API version (game-update style: check version → download if changed).
  static const String versionPath = '/api/standee/version';
  static String get versionUrl => '$baseUrl$versionPath';

  // ============ SDP (Special Dynamic Poster) APIs ============

  /// WebSocket endpoint cho Weather realtime
  static const String weatherWsPath = '/ws/topic/weather';
  static String get weatherWsUrl => '$wsBaseUrl$weatherWsPath';

  /// REST API cho Air Quality
  static const String airQualityPath = '/api/weather/air-quality';
  static String get airQualityUrl => '$baseUrl$airQualityPath';

  /// REST API cho Speed Test
  static const String speedTestPath = '/api/speedtest/all';
  static String get speedTestUrl => '$baseUrl$speedTestPath';

  /// REST API cho Standee Info
  static const String standeeInfoPath = '/api/standee';
  static String get standeeInfoUrl => '$baseUrl$standeeInfoPath';

  // ============ SDP Configuration ============

  /// Duration mặc định của SDP (giây) - sẽ được override bởi API displayDuration
  static const int sdpDurationSeconds = 40;

  /// Asset path cho SDP background
  /// - tetve.jpg: Theme sáng (Tết vè) - phù hợp ban ngày
  /// - tetden.jpg: Theme tối - phù hợp ban đêm
  static const String sdpBackgroundLight = 'assets/tetve.jpg';
  static const String sdpBackgroundDark = 'assets/tetden.jpg';

  /// Background mặc định cho SDP (Tết vè)
  static const String sdpDefaultBackground = sdpBackgroundLight;
}

File: src/core/constants/apiConstants.ts
const baseUrl = 'http://10.10.115.20:9094';
const wsBaseUrl = 'ws://10.10.115.20:9094';
const standeeId = '7bbb9af3-95a7-42af-8805-6023e60a3bdc';

export const apiConstants = {
  baseUrl,
  wsBaseUrl,
  standeeId,
  playlistPath: `/api/poster/${standeeId}/withActivePosters`,
  get playlistUrl() {
    return `${baseUrl}${this.playlistPath}`;
  },
  versionPath: '/api/standee/version',
  get versionUrl() {
    return `${baseUrl}${this.versionPath}`;
  },
  weatherWsPath: '/topic/weather',
  /** Raw WebSocket (nếu server dùng ws://host/ws/websocket). */
  weatherWsConnectPath: '/ws/websocket',
  get weatherWsConnectUrl() {
    return `${wsBaseUrl}${this.weatherWsConnectPath}`;
  },
  /** SockJS endpoint (nếu server trả /ws/390/{session}/websocket thì dùng cái này). */
  weatherWsSockJsPath: '/ws',
  get weatherWsSockJsUrl() {
    return `${baseUrl}${this.weatherWsSockJsPath}`;
  },
  /** true = dùng SockJS, false = dùng raw WebSocket. */
  weatherUseSockJs: true,
  airQualityPath: '/api/weather/air-quality',
  weatherPath: '/api/weather',
  speedTestPath: '/api/speedtest/all',
  standeeInfoPath: '/api/standee',
  /** GET /api/poster/{standeeId}/withActivePosters */
  getPosterPath(id: string) {
    return `/api/poster/${id}/withActivePosters`;
  },
  /** Nhạc nền: GET /audio/nhacbuoisang.mp3 */
  audioPath: '/audio/nhacbuoisang.mp3',
  get audioUrl() {
    return `${baseUrl}${this.audioPath}`;
  },
  sdpDurationSeconds: 40,
  sdpDefaultBackground: 'assets/tetve.jpg',
  /** Bật/tắt poster trong slideshow: true = SDP → poster → ...; false = chỉ hiển thị SDP lặp. */
  slideshowShowPosters: true,
  /** Bật/tắt video (catoon.mp4) trong slideshow: true = có slide video; false = không có. */
  slideshowShowVideo: false,
  /** Test hiệu ứng: 'rain' | 'storm' | 'cloudy' | 'flowers' | null. 'storm' = bão mưa to + sét. */
  weatherEffectOverride: null as 'rain' | 'storm' | 'cloudy' | 'flowers' | null, 
} as const;
